homeassistant/components/notify/yessssms.py

```diff
@@ -38,14 +39,37 @@ def __init__(self, username, password, recipient):
         from YesssSMS import YesssSMS
         self.yesss = YesssSMS(username, password)
         self._recipient = recipient
+        _LOGGER.debug(
+            "initialized; library version: %s", self.yesss.version())
 
     def send_message(self, message="", **kwargs):
         """Send a SMS message via Yesss.at's website."""
+        if self.yesss.account_is_suspended():
+            # only retry to login after HASS was restarted with (hopefully)
+            # new login data.
+            _LOGGER.error(
+                "Account is suspended, cannot send SMS. "
+                "Check your login data and restart Home Assistant")
+            return
         try:
             self.yesss.send(self._recipient, message)
-        except ValueError as ex:
-            if str(ex).startswith("YesssSMS:"):
-                _LOGGER.error(str(ex))
-        except RuntimeError as ex:
-            if str(ex).startswith("YesssSMS:"):
-                _LOGGER.error(str(ex))
+        except self.yesss.NoRecipientError as ex:
+            _LOGGER.error(
+                "You need to provide a recipient for SMS notification.",
+                exc_info=ex)
+        except self.yesss.EmptyMessageError as ex:
+            _LOGGER.error(
+                "Cannot send empty SMS message: %s", ex)
```

Comment:
I should have been more clear with my last comment, but let's pass the full exception with `exc_info=ex` for all of these logs lines.

Reply:
I don't see the use in that. It messes up the log, and the cause is clear. It's expected, defined behavior.
eg. EmptyMessageError is raised when the user attempts to send an empty message. It is caught by the lib, and not even passed on to the online service.
also the LoginError: the credentials are wrong and need to be changed.

NoRecipientError made a little sense to log the whole exeception, but it would be a configuration error. Stil one could argue that the log is made unreadable when these errors are caught.

Reply:
can we do another line, with `_LOGGER.debug` that does pass the full exception? to keep the default log nicer?

Reply:
Ah, good catch. I agree the empty message error doesn't need to include the full exception. It's pretty straightforward on its own. We should add the object though for the other errors which are unexpected, since the information is useful for debugging.

Also if there are errors which will happen frequently, a separate debug with the full exception is OK too.